plugins/lookup/etcd_patroni_nodes.py

```diff
@@ -0,0 +1,180 @@
+                sby = etcd_standbys[k]
+
+                if sby['upstream_node_private_ip'] in etcd_primary_map:
+                    upstream_private_ip = sby['upstream_node_private_ip']
+                    primary_private_ip = etcd_primary_map[upstream_private_ip]
+                    etcd_primary_map[sby['private_ip']] = primary_private_ip
+                    etcd_clusters[primary_private_ip].append(sby)
+                    del(etcd_standbys[k])
+
+            # Case when at least one host has not been handled in this loop
+            # iteration.
+            if etcd_standbys_len == len(etcd_standbys.keys()):
+                raise AnsibleError(
+                    "Inventory error with the following standbys nodes %s. "
+                    "Upstream node is not configured or not found"
+                    % [s for s in etcd_standbys.keys()]
+                )
+
+            etcd_standbys_len = len(etcd_standbys.keys())
+
+        # Append witness nodes into the right etcd_clusters item, based on
+        # witness's upstream node.
+        while etcd_witnesses_len != 0:
+
+            for k in list(etcd_witnesses.keys()):
+                wit = etcd_witnesses[k]
+
+                if wit['upstream_node_private_ip'] in etcd_primary_map:
+                    upstream_private_ip = wit['upstream_node_private_ip']
+                    primary_private_ip = etcd_primary_map[upstream_private_ip]
+                    etcd_primary_map[wit['private_ip']] = primary_private_ip
+                    etcd_clusters[primary_private_ip].append(wit)
+                    del(etcd_witnesses[k])
+
+            # Case when at least one host has not been handled in this loop
+            # iteration.
+            if etcd_witnesses_len == len(etcd_witnesses.keys()):
+                raise AnsibleError(
+                    "Inventory error with the following witness nodes %s. "
+                    "Upstream node is not configured or not found"
+                    % [s for s in etcd_witnesses.keys()]
+                )
+
+            etcd_witnesses_len = len(etcd_witnesses.keys())
+
+
+        if node_private_ip in etcd_primary_map:
+            # Current node is part of one of the SR clusters found
+            return etcd_clusters[etcd_primary_map[node_private_ip]]
+        else:
+            primary_private_ips = list(etcd_clusters.keys())
+            # If the current node is not part of any ETCD cluster found, but,
+            # only one ETCD cluster has been found, then we return this ETCD
+            # cluster because there is no doubt.
+            if len(primary_private_ips) == 1:
+                return etcd_clusters[primary_private_ips[0]]
+            else:
+                raise AnsibleError(
+                    "Unable to find the ETCD cluster topology because multiple "
+                    "ETCD clusters were found and this current node does not "
+                    "appear to be part of any of them"
+                )
```

Comment:
For etcd, we shouldn't apply the same logic used for building PSR cluster topology. We only want to setup etcd on the hosts where the `etcd` hostvar is set to true.

Reply:
It was fixed as per our discussion and utilization of etcd_nodes plugin.